### ISSUE TITLE
Redo Vore System

### ISSUE THREAD

**Issue by Arokha** [SpadesNeil/VOREstation#601]
There are several features the vore system could do with:
- Better management of prey in preds who have something happen (die, del(), digested)
- Better procs for doing specifically digestion damage
- Resisting digestion by wearing armor, based on bio armor stat
- Custom internal spaces with properties (name, description, sounds, digestion, messages)
- Redone 'inside' panel (probably renamed...) to show/edit the custom spaces, mobs, and items 'inside'
- Allow preds to specify digestion strength (within limits...)

More notes (04/18/16 9:39pm EDT):
- All mobs should be able to have bellies, simple or carbon or silicon
- All mobs should be able to customize them, but, also have them pre-defined in code for simple_mobs and such that need to use them for their routines
- All the custom text should be saved in the save files
- 10 bellies should be enough, I'd think/hope

**Comment by SpadesNeil** [comment]
- Allow preds to specify digestion strength (within limits...)

This is the only one I have a problem with. The current system allows for about 10 minutes of RP time before you're gurgled, so even if the other player has digestion turned on immediately, there's ample time to roleplay. Any weaker digestion and the passive healing is too strong and 10 minutes suddenly jumps to a lot more which varies heavily depending how much nutrition the victim has. Any stronger and it cuts severely on the RP time without the predator toggling digestion.

What I recommend instead is that the predator be given the ability to watch the player's health variable and get flavor text according to how healthy their victim is.

**Comment by Arokha** [comment]
It's not really to let people decide how long their RP is, it's more for people who want to digest their prey quickly at the end of the RP, so they don't need to try to 'predict' when 10 minutes will be. I know most of my RPs are mostly about the eating and stuff, and digestion only occupies the last bit of the RP with the prey still alive. Always annoying to try to juggle it. I usually end up VV'ing my prey to balance it. Instead of giving preds a pseudo-VV thing to show health though, since that doesn't fix my issue of needing to 'start digestion before it starts in the RP', I'd say just let people crank up digestion.

**Comment by SpadesNeil** [comment]
Well that's why I recommended the health indication. We could even set it up to hold the player's health around a certain point so throughout the RP the prey is being immersed by the damage being caused to them. I dislike it when I'm undigested for an entire RP and then only at the very end do they turn on digestion and kill my character.

There's the problem of what if something happens? What if they have to spit me up? It annoys me when I've been spat up totally unscathed despite the RP saying I've been stewing for some time in bubbling acids only for Medbay to have little to nothing to do and heal me for, or at the very least, nothing for the medibot to do anything for.

I propose a compromise. Health is tied directly to the damage sustained by a player. When health reaches -100 the player always dies. Health is calculated based on your current brute, burn, toxin, etc damages. I want to see a system where the predator can roughly gauge the health of their prey.

Examples:
80-100 health | Your prey feels fully intact.
79 - 40 health | Your prey feels slick and weaker to the touch.
39 - 0 health | Your prey feels like they're beginning to soften up.
0 - (-60) | Your prey's movement is slow and groggy
(-61) - (-100) | Your prey feels like they're about to break down into mush.

I once tested to see at what point you consistently lose consciousness but I lost the data on that somewhere.

**Comment by Treasa729** [comment]
Why not just make it so that the pred can toggle off health regeneration for the prey instead? I think a variable digestion time is a really good idea, people's scenes take different amounts of time, and having a slower or faster digestion should be the player's choice. I definitely think the pred should be given an indictation of the prey's health, but as it is, having to juggle the digestion on and off is just incredibly annoying from a predator's position. You say digestion as it is now gives 10 minutes of solid RP time, but for the people who enjoy more verbose posts, 10 minutes is nothing at all, and if people want faster digestion, I don't see why that shouldn't be allowed either.  

As it is now, the mechanics are something clunky that we have to work with in order to do the RP, letting people control the speed they digest things at allows for the mechanics to compliment the RP, not limit it. The locking it at certain points is good suggestion as well, and possibly making it a thing in the inside panel beside each prey where you have their health shown, and how low you want it to be able to go could work, but I feel that having that as the only solution limits the RP. 

Let's say, for example, that the pred is doing a scene with multiple people, and they would each be at different points in digestion because of the scene, being able to speed up or slow down the rate of digestion works great for facilitating that sort of RP. You don't have to just wait it out, and digest them both simultaneously because the digestion of 10 minutes is too short to have on from the start in a multi-person scene, you can just set the digestion to be much slower, and allow the scene to play out naturally.

In a scene with two verbose RPers, there can often end up being 10 minutes between each post, alone, and not being able to control that to encourage that sort of RP is important too. There are also people who _like_ very long digestion. They want to be within their predator, digesting for 30 minutes, an hour, even, as the predator goes about their business, rather than just having the 10 minutes, and that's it. If you want to do long digestion, either juggle it on and off, leave it off, or go somewhere else.

Some people like shorter or longer digestion, and you said 10 minutes was ample time, but it may not be for all people, your prefs aren't everyone's. Some people are fast typers, others are slow typers, some like to take a while to revise, reword their posts, others just want to progress the RP as fast as they can. Your style of RP doesn't account for everyone's, and I know there are some people who make posts that have to made in anywhere from 5 - 15 posts in game because they are very verbose in their writing, and they like to make it as descriptive as they possibly can. That sort of posting doesn't happen in 10 minutes, heck, it doesn't happen in 30. Much longer isn't necessarily bad, because there are people who like it to be much longer. There are also people who like to just make a few quick posts back and forth, and are done the RP in 5 minutes, then just waiting for the digestion to go afterward.

TL:DR I feel that the mechanics should be made to compliment as many people's stlyes of RP as they can, and a controllable digestion speed facilitates that. This, combined with the locking system is good, but they could each be used in different scenarios, and either method, or both would be more appealing to different people.

**Comment by SpadesNeil** [comment]
> Why not just make it so that the pred can toggle off health regeneration for the prey instead?

Because that's how the medical system works and it's hard coded in. I could change it but this carries risk of breaking things and I have absolutely no idea where in the code that's even defined.

It's not a good idea so we're not going to mess with that.

**Comment by Arokha** [comment]
Almost done with this.
